source/MaterialXTest/MaterialXGenShader/GenShaderUtil.cpp

```diff
@@ -861,7 +861,10 @@ void ShaderGeneratorTester::validate(const mx::GenOptions& generateOptions, cons
                         }
                         else
                         {
-                            path = path / (elementName + "." + getFileExtensionForLanguage(_shaderGenerator->getLanguage()));
+                            path = path / (elementName + "."  
+                                + _shaderGenerator->getTarget() 
```

Comment:
Add "target" to file name to avoid naming duplicates between variants on a language.